src/tools/crawler/rpc.rs

```diff
@@ -37,18 +37,26 @@ pub async fn initialize_rpc_server(rpc_addr: SocketAddr, rpc_context: RpcContext
 fn create_rpc_module(rpc_context: RpcContext) -> RpcModule<RpcContext> {
     let mut module = RpcModule::new(rpc_context);
     module
-        .register_method("getmetrics", |params, rpc_context| {
+        .register_method("getmetrics", |_params, rpc_context| {
+            Ok(rpc_context.0.lock().unwrap().clone())
+        })
+        .unwrap();
+    module
+        .register_method("dumpmetrics", |params, rpc_context| {
             let report_params = params.parse::<ReportParams>()?;
             if let Some(path) = report_params.file {
                 let content = serde_json::to_string(rpc_context.0.lock().unwrap().deref())?;
-                let length = content.len();
+                // Wrap our NetworkSummary in a JSON-RPC response envelope
+                let response =
+                    "{\"jsonrpc\":\"2.0\",\"result\":".to_owned() + &content + ",\"id\":0}";
+                let length = response.len() as i32;
                 // TODO: consider some checks against directory traversal
-                if let Err(e) = fs::write(path, content) {
+                if let Err(e) = fs::write(path, response) {
                     warn!("Unable to write to file: {}", e);
```

Comment:
If path is not right this function returns -1, but if path is ok but cannot write to file we'll return the `length` just as if writing was successful - is that correct?

Reply:
good find, yes, it's a third return branch by itself.